src/Microsoft.Data.Analysis/DataFrameRow.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Data.Analysis
+{
+    /// <summary>
+    /// A DataFrameRow is a collection of values that represent a row in a <see cref="DataFrame"/>.
+    /// </summary>
+    public class DataFrameRow : IEnumerable<object>
+    {
+        private readonly DataFrame _dataFrame;
+        private readonly long _rowIndex;
+        internal DataFrameRow(DataFrame df, long rowIndex)
+        {
+            Debug.Assert(rowIndex < df.Columns.RowCount);
+            _dataFrame = df;
+            _rowIndex = rowIndex;
+        }
+
+        /// <summary>
+        /// Returns an enumerator of the values in this row.
+        /// </summary>
+        public IEnumerator<object> GetEnumerator()
+        {
+            foreach (DataFrameColumn column in _dataFrame.Columns)
+            {
+                yield return column[_rowIndex];
+            }
+        }
+
+        /// <summary>
+        /// An indexer to return the value at <paramref name="index"/>.
+        /// </summary>
+        /// <param name="index">The index of the value to return</param>
+        /// <returns>The value at this <paramref name="index"/>.</returns>
+        public object this[int index]
+        {
+            get
+            {
+                return _dataFrame.Columns[index][_rowIndex];
+            }
+            set
+            {
+                _dataFrame.Columns[index][_rowIndex] = value;
+            }
+        }
+
+        /// <summary>
+        /// The number of values in this row.
+        /// </summary>
+        public int Count => _dataFrame.Columns.Count;
```

Comment:
`DataFrameRow.Count` is a little odd to me. I would if this should be `ValuesCount` or something. 

Thoughts on a different name here?

Reply:
I can't think of a strong choice here. `ValuesCount` sounds good to me. I checked to see if IDataView.DataViewRow has an equivalent, it doesn't.

Reply:
Or maybe we drop `Count` here, and the user can go to the `Columns.Count` instead?

Reply:
https://apisof.net/catalog/System.Data.DataRow also doesn't have a `Count` property. How about we drop it for now, and add it later if we see a need.

Reply:
Hmmm, that's not a bad idea. If they include Linq, they'll be able to get the number of values anyway.

Reply:
Yup, dropping.

Reply:
It would mean it's not straightforward to mutate the values in a `DataFrameRow` though. An indexer exists on `DataFrameRow`, but to use the right index, users would need to know the number of values in the row. The foreach enumeration is readonly.

Reply:
the user can go to the Columns.Count instead